internal/util/util_test.go

```diff
@@ -119,3 +119,32 @@ func Test_ExtractPath(t *testing.T) {
 		})
 	}
 }
+
+func Test_CompareStructWithoutNumKind(t *testing.T) {
+	testCases := []struct {
+		name               string
+		a                  map[string]interface{}
+		b                  map[string]interface{}
+		expectedComparison bool
+	}{
+		{
+			name:               "equal structs - same number type",
+			a:                  map[string]interface{}{"number": 1},
+			b:                  map[string]interface{}{"number": 1},
+			expectedComparison: true,
+		},
+		{
+			name:               "equal structs - different number type",
+			a:                  map[string]interface{}{"number": 1},
+			b:                  map[string]interface{}{"number": float64(1)},
+			expectedComparison: true,
```

Comment:
would you mind adding at least one more case in here, which includes the uint type?

Reply:
Added test cases for all types